### PR TITLE
Prefix imports with workspace/package name if possible

### PR DESCRIPTION
**Pull request by mariusgrigoriu**
Fixes issue #55 

This is a cleaned up version of the commit from #46 without the bad stuff.

### CODE REVIEW

**Comment by seh** [comment]
Who can review this change? @davidzchen was the previous author of the code changed here.

File: examples/BUILD [hunk 1]
```diff
@@ -116,3 +116,23 @@ jsonnet_to_json_test(
     yaml_stream = 1,
     golden = "yaml_stream_golden.yaml",
 )
+
+jsonnet_to_json(
```
Comment (bshi) [inline examples/BUILD]: Please remove this and add a golden file to the jsonnet_to_json_test below.

(See https://github.com/bazelbuild/rules_jsonnet/blob/master/examples/BUILD#L3)

**Comment by bshi** [comment]
I'll try to track down a skylark domain expert to review this.

**Comment by seh** [comment]
I tested this patch, and find that it still doesn't address a problem with the "imports" attribute picked up in transitive dependencies on `jsonnet_library` targets. With a target in package "a/b/c" and an "import" entry of "../../x," when that first target is mentioned directly or transitively by other targets, the resulting Jsonnet invocation includes multiple `-J` flags with nonsensical values like the following:
- _a/b/c/a/b/c/../../x_
- _a/b/c/a/b/c/a/b/c/../../x_

Sometimes I'm able to get the intended `-J` flag value of _/a/b/c/../../x_, but not when the `jsonnet_library` target is at least two levels away in my dependency graph.

It looks like each time we incorporate a transitive "import" entry, we prefix it with the package path of the dependency, ignoring that that package path may have already been added in an earlier consideration of the dependency. That is, "import" entries pass through the `_setup_deps` function first in each `jsonnet_library` target that depends on the target that first defines them, then once more in any `jsonnet_to_json` targets that depend on those libraries.

_[Time passes...]_

I found a fix that appears to work for my projects: Only call `_add_prefix_to_imports` from `_jsonnet_library_impl` when collecting the `imports` value, and **not** in `_setup_deps`:
```python
imports = depinfo.imports + _add_prefix_to_imports(ctx.label, ctx.attr.imports)
```
That leaves less for `_setup_deps` to accomplish.

@mariusgrigoriu, I'd appreciate your thoughts on these findings.

**Comment by seh** [comment]
Please see #77 for another take on this patch.

**Comment by Globegitter** [comment]
This has been replaced by #77

**Pull request closed without merging**